tests/e2e/functional_test.go

```diff
@@ -88,6 +88,11 @@ func (s *FunctionalSuite) TestNumaflowGitSync() {
 	w.Expect().ResourcesDontExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"http-pipeline"})
 	w.Expect().CheckCommitStatus()
 
+	w.PushToGitRepo("numaflow/initial-commit", []string{"sample_pipeline.yaml", "isbsvc_jetstream.yaml"}, true).Wait(30 * time.Second)
+	w.Expect().ResourcesDontExist("numaflow.numaproj.io/v1alpha1", "pipelines", []string{"simple-pipeline"})
```

Comment:
You may want to delete the pipeline first? but it may not matter.

Reply:
I think it's okay like this, as long as both are deleted before the controller